doc/Language/control.pod6

```diff
@@ -39,24 +39,38 @@ executed.
     { say 2; say 3 };         # OUTPUT: «2␤3␤»
     say 4;                    # OUTPUT: «4␤»
 
-Unless it stands alone as a statement, a block simply creates a closure. The
-statements inside are not executed immediately. Closures are another topic
-and how they are used is explained
-L<elsewhere|/language/functions#Blocks_and_lambdas>. For now it is just
-important to understand when blocks run and when they do not:
+This might be useful if you are using an editor that offers code folding,
+so you can use these blocks to add visual sections to your code. 
+Blocks also create their own namespace, so can be used for
+scoping.
+ 
+    my $number = 1;
+    print $number ~ ', ';
+    { 
+        my $number = 2;
+        print $number ~ ', ';
+            {
+                my $number = 3;
+                print $number ~ ', ';
+            }
+    }
+    say $number;
+    # OUTPUT: «1, 2, 3, 1␤»
+
+However, if a block is used in an expression, then it doesn't run immediately. 
```

Comment:
Again, it does not because a block defines a function; it does not run in the same way a function does not run. Maybe you should explain what's the expression context here, and why in the first case it's directly called (as if it had a () behind) and in the second case it's in expression context, so it's not invoked. But the examples weren't too good to start with. Showing intent in this case would mean assigning a block to some other variable, and using it as Callable code.